### DISCUSSION TITLE
Primer Brand v0.24 - Technical Preview

### DISCUSSION THREAD

**Discussion by rezrah** [primer/brand#395] [Announcements]
![Introductory banner with a blue/purple background gradient showing a foreground browser chrome featuring a new landing page design using site refactored styles. Also features a monamoji with heart eyes with a heading Technical Preview for 0.24.](https://github.com/primer/brand/assets/13340707/5d36ee70-77fd-436c-aeac-cb1cb73a8d7d)

The Primer Brand team is excited to preview a host of improvements and new features that are shipping in the next version of `@primer/react-brand`.

**This technical preview will cover:**

- [What's new](#summary)
- [How to try it yourself today](#try-it)
- [Release timeline](#schedule)

## Overview
Primer Brand libraries (both React and Figma) are currently undergoing a major visual refactoring process, whereby foundational aspects to the system - like typography - are being completely overhauled. This has inevitably led to wholesale visual changes across all components. 

These changes are being rolled out to help make our patterns more accessible and aligned with our future art direction. 

> [!IMPORTANT]
> This technical preview is intended to provide advance notice for various upcoming breaking changes. Please read carefully to understand how this may impact you.

## 🔧  <a name="try-it">Try it today</a>
You can try `v0.24.0` RC today

```
npm install @primer/react-brand@next
```
This release candidate is available for preview and feedback purposes only. **We don’t recommend using this in production**.

## 📆  <a name="schedule"> Release schedule</a>
<table>
<tr>
<th> 15 August</th> <td> Public beta for `v0.24.0` on the "next" release channel</td>
</tr>
<tr>
<th> 17 August</th> <td> Close public beta</td>
</tr><tr>
<th> 18 August</th> <td> General availability on the "Latest" release channel</td>
</tr>
</table>

##  ✨  New in `0.24.0`
- River breakout component (coming soon)
- Emphasis support in large testimonials (coming soon)
- Video player (coming soon)

#### <a id="summary">Summary of changes </a>

`v0.24.0` will contain a handful of visual and API breaking changes.

Are you using any of the components below? Skip to the relevant section below to learn about what's changing.

Skip to:

- [`Heading`](#heading-component)
- [`Text`](#text-component)
- [`River`](#river-component)
- [`SectionIntro`](#sectionintro-component)
- [`Label`](#label-component)

### <a href="#heading-component" name="heading-component">🔗 Heading</a>
![banner showing new type scale](https://github.com/primer/brand/assets/13340707/b086f6fe-f083-435b-8163-68a8e6f3b156)

- `extra-bold` font weight no longer default.
- New `display` size added, which is the new largest heading size.
- New `subhead-large` and `subhead-medium` sizes added, which are the new smallest heading sizes.
- All headings now use either `bold` or `semi-bold`.

<table>
<tr>
<th> Before</th> <th> After</th>
</tr>
<tr>
<td valign="top">

```jsx
import { Heading } from '@primer/react-brand';

<>
  <Heading size="1">1</Heading>
  <Heading size="2">2</Heading>
  <Heading size="3">3</Heading>
  <Heading size="4">4</Heading>
  <Heading size="5">5</Heading>
  <Heading size="6">6</Heading>
</>
```

 </td>
<td valign="top">

```jsx
import { Heading } from '@primer/react-brand';

<>
  <Heading size="display">Display</Heading>
  <Heading size="1">1</Heading>
  <Heading size="2">2</Heading>
  <Heading size="3">3</Heading>
  <Heading size="4">4</Heading>
  <Heading size="5">5</Heading>
  <Heading size="6">6</Heading>
  <Heading size="subhead-large">Subhead large</Heading>
  <Heading size="subhead-medium">Subhead medium</Heading>
</>
```

</td>
</tr>
</table>

### <a href="#text-component" name="text-component">🔗  Text</a>
![banner showing three new standardised sizing, with 16px being the default](https://github.com/primer/brand/assets/13340707/f6ea0c5f-f156-45df-9777-3fc045c12740)

- Base text size now `200` (previously `300`), which corresponds to `16px` (previously `18px`).
- All text sizes now use 450 weight instead of 400

<table>
<tr>
<th> Before</th> <th> After</th>
</tr>
<tr>
<td valign="top">

```jsx
import { Text } from '@primer/react-brand';

<Text size="300">Prev. default text size</Text>
```

 </td>
<td valign="top">

```jsx
import { Text } from '@primer/react-brand';

<Text size="200">New default text size</Text>
```

</td>
</tr>
</table>

### <a href="#river-component" name="river-component">🔗  River</a>
![banner showing a river with updated typographic styling](https://github.com/primer/brand/assets/13340707/475b0d1b-19db-48f0-b0cf-200acb7c2022)

- River visuals now automatically apply rounded corners
- Shadows are now turned off by default
- Text size now 16px (new default)
- Link size reduced to `medium`
- Smaller heading size

Restore shadows and non-rounded corners using `rounded` and `hasShadow` props:

```jsx
import { River } from '@primer/react-brand';

<River>
  <River.Visual rounded={false} hasShadow={true}>
    <img />
  </River.Visual>
  <River.Content />
</River>
```

### <a href="#sectionintro-component" name="sectionintro-component">🔗  Section intro</a>
![banner showing section intro component in new sizing, with a label at the top](https://github.com/primer/brand/assets/13340707/4cc60ff0-6553-4e47-ba1d-94e447f49d8e)

- Section intro heading now 40px by default
- Label updated to `small` size 
- Description text now 16px and default text color
- Duotone support added using `<em>`

```jsx
import { SectionIntro } from '@primer/react-brand';

<SectionIntro>
  <SectionIntro.Heading>
    <em>Expressive headline</em> about an exclusive set of features.
  </SectionIntro.Heading>
</SectionIntro>
```

### <a href="#label-component" name="label-component">🔗  Label</a>
![banner showing various labels, specifically the small and default sizes which are now the same dimensions](https://github.com/primer/brand/assets/13340707/0ae84a28-6f1f-4dcd-a4bc-03452ec4a681)

- `small` size has been removed, leaving only `medium` and `large` options.

### <a href="#faq-component" name="faq-component">🔗  FAQ</a>
![banner showing new FAQ group styling](https://github.com/primer/brand/assets/13340707/b01bd11f-1572-4a84-a489-c71eff937deb)

- FAQ heading weights reduced
- FAQ group spacing bug fixed 
- FAQ.Heading `size` prop values have been replaced with Heading defaults for more granular control and consistency.

<table>
<tr>
<th> Before</th> <th> After</th>
</tr>
<tr>
<td valign="top">

```jsx
<>
  <FAQ.Heading size="medium">medium (default)</FAQ.Heading>
  <FAQ.Heading size="large">large</FAQ.Heading>
</>
```

 </td>
<td valign="top">

```jsx
<>
  <FAQ.Heading size="2">2</FAQ.Heading>
  <FAQ.Heading>new default ("3")</FAQ.Heading>
  <FAQ.Heading size="4">4</FAQ.Heading>
  <FAQ.Heading size="5">5</FAQ.Heading>
  <FAQ.Heading size="6">6</FAQ.Heading>
</>
```

</td>
</tr>
</table>

## Reporting issues

Found a bug or problem while trying `v0.24.0`? Please [raise an issue here](https://github.com/primer/brand/issues/new?created_from_discussion_number=395).

Thanks for reading this technical preview. We look forward to shipping this release soon 🚀

**Comment by Agony23** [comment]
[☢️Terms]()
